_docs/tasks/integrating-services-into-istio.md

```diff
@@ -1,36 +1,224 @@
+echo.yaml.
 
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: echo
+  labels:
+    app: echo
+spec:
+  ports:
+  - port: 80
+    targetPort: 8080
+    name: http
+  selector:
+    app: echo
+---
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: echo
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: echo
+    spec:
+      containers:
+      - name: echo
+        image: gcr.io/google_containers/echoserver:1.4
+        ports:
+        - containerPort: 8080
+---
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: busybox
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: busybox
+    spec:
+      containers:
+      - name: busybox
+        image: radial/busyboxplus:curl
+```
 
-## Understanding ...
+[Kubernetes Services](https://kubernetes.io/docs/concepts/services-networking/service/)
+are required for properly functioning Istio service. Service ports
+must be named and these names must begin with _http_ or _grpc_ prefix
+to take advantage of Istio's L7 routing features, e.g. `name: httpFoo`
+is good. Services with non-named ports or with ports that do not have
+a _http_ or _grpc_ prefix will be routed as L4 traffic.
 
-Here's an interesting thing to know about the steps you just did.
+Submit a YAML resource to API server with injected Envoy sidecar. Any
+one of the following methods will work.
 
+```bash
+kubectl apply -f <(istioctl kube-inject -f echo.yaml)
+```
 
-## What's next
-* Learn more about [this](...).
-* See this [related task](...).
+Make a request from the client (busybox) to the server (echo).
+
+```bash
+$ CLIENT=$(kubectl get pod -l app=busybox -o jsonpath='{.items[0].metadata.name}')
+$ SERVER=$(kubectl get pod -l app=echo -o jsonpath='{.items[0].metadata.name}')
+
+$ kubectl exec -it ${CLIENT} -c echo -- curl echo:80 | grep x-request-id
+x-request-id=a641eff7-eb82-4a4f-b67b-53cd3a03c399
+```
+
+Verify traffic is intercepted by the Envoy sidecar. Compare
+`x-request-id` in the HTTP response with the sidecar's access
+logs. `x-request-id` is random. The IP in the inbound request logs is
+the echo pod's IP.
+
+Outbound request on client pod's proxy.
+
+```
+$ kubectl logs ${CLIENT} proxy | grep a641eff7-eb82-4a4f-b67b-53cd3a03c399
+[2017-05-01T22:08:39.310Z] "GET / HTTP/1.1" 200 - 0 398 2 0 "-" "curl/7.47.0" "a641eff7-eb82-4a4f-b67b-53cd3a03c399" "echo" "127.0.0.1:8080"
+```
+
+Inbound request on server pod's proxy.
+
+```
+$ kubectl logs ${SERVER} proxy | grep a641eff7-eb82-4a4f-b67b-53cd3a03c399
+[2017-05-01T22:08:39.310Z] "GET / HTTP/1.1" 200 - 0 398 3 3 "-" "curl/7.47.0" "a641eff7-eb82-4a4f-b67b-53cd3a03c399" "echo" "10.4.180.7:8080"
+```
+
+
+The Envoy sidecar does _not_ intercept container-to-container traffic
+within the same pod when traffic is routed via localhost. This is by
+design.
 
+```bash
+$ kubectl exec -it ${SERVER} -c echo -- curl localhost:8080 | grep x-request-id
+```
+
+## Understanding what happened
+
```

Comment:
Are we following this title convention everywhere? Can we retitle this as "Under the hood"?